### PR TITLE
Support CentOS 8 RPMs

### LINKED ISSUE #5319: RPMs only target CentOS 7
## I'm submitting a ...

-  new feature / enhancement request

## Traffic Control components affected ...

-  Grove
-  Traffic Monitor
-  Traffic Ops
-  Traffic Ops ORT
-  Traffic Portal
-  Traffic Router
-  Traffic Stats

## Current behavior:

Right now, all of our RPMs are targeted at CentOS 7, whose *Full Updates* support [ends in Q4 2020](https://wiki.centos.org/About/Product).

## New behavior:

We should support CentOS 8 as a build target, as using CentOS 8 instead of CentOS 7 is inevitable.

## Minimal reproduction of the problem with instructions:

1. Build the Apache Traffic Control RPMs
2. Notice that their filenames all end with `.el7.x86_64.rpm` or `.el7.src.rpm`

### PR DESCRIPTION
**Pull request by zrhoffman**
## What does this PR (Pull Request) do?

- [x] This PR fixes #5319
- Adds the option to build for either CentOS 7 or CentOS 8. CentOS 7 is the default, except for CDN in a Box, which uses CentOS 8 RPMs by default.
- Upgrades the builder Docker images to CentOS 8, although they can still be built for CentOS 7.

This PR builds off of @hbeatty's CentOS 8 progress from #5137.

## Which Traffic Control components are affected by this PR?

- CDN in a Box
- Documentation
- Build system

## What is the best way to verify this PR?

1. Check that Traffic Control works with RPMs built on and targeting CentOS 8
    ```shell script
   make build-builders -j6
   docker-compose build --parallel
   docker-compose up
    ```

2. Check that Traffic Control works with RPMs built on and targeting CentOS 7 
    ```shell script
   export RHEL_VERSION=7
   make build-builders -j6
   docker-compose build --parallel
   docker-compose up
    ```

## The following criteria are ALL met by this PR

- [x] Existing tests are sufficient, no additional tests are unnecessary
- [x] This PR includes documentation
- [x] This PR includes an update to CHANGELOG.md
- [x] This PR includes any and all required license headers
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

## Additional Information

Note that only Go-based projects contain binaries that `elfdeps` checks the linking for. The Go binaries have been statically linked since #4758, so that check does not find any dynamic section in the binaries, and we can use CentOS 8 builders to build RPMs targeting either CentOS 7 (the default for now) or CentOS 8.

### CODE REVIEW

File: infrastructure/cdn-in-a-box/Makefile [hunk 1]
```diff
@@ -79,6 +93,16 @@ native: all
...
+PKG_FLAGS += -p
+ifneq ($(MAKECMDGOALS),pull-builders)
+MAKECMDGOALS := $(filter-out pull-builders,$(MAKECMDGOALS))
+pull-builders: $(MAKECMDGOALS)
+else
+pull-builders: all
+endif
+endif
+
 #.PHONY: native
```
[the remark this answers is not in the document]
Comment (zrhoffman) [reply infrastructure/cdn-in-a-box/Makefile:106]: This is the `pull-builders` target, the other one is the `build-builders` target.

**Comment by zrhoffman** [comment]
Rebased to fix merge conflicts

File: infrastructure/cdn-in-a-box/Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (ocket8888) [reply infrastructure/cdn-in-a-box/Makefile:106]: No, I'm talking about `#.PHONY: native`

[the remark this answers is not in the document]
Comment (zrhoffman) [reply infrastructure/cdn-in-a-box/Makefile]: Gotcha, removed in 899e3f1cde

**Review by ocket8888 — APPROVED** [review]
Works

**Pull request merged**